storage.go

```diff
@@ -2,35 +2,181 @@ package ftp
 
 import (
 	"context"
+	"errors"
 	"io"
+	"path/filepath"
 
+	"github.com/beyondstorage/go-storage/v4/pkg/iowrap"
 	. "github.com/beyondstorage/go-storage/v4/types"
+	"github.com/jlaffaye/ftp"
+	mime "github.com/qingstor/go-mime"
 )
 
+type listDirInput struct {
+	rp  string
+	dir string
+
+	started           bool
+	continuationToken string
+	objList           []*ftp.Entry
+	counter           int
+}
+
 func (s *Storage) create(path string, opt pairStorageCreate) (o *Object) {
-	panic("not implemented")
+	if opt.HasObjectMode && opt.ObjectMode.IsDir() {
+		o = s.newObject(false)
+		o.Mode = ModeDir
+	} else {
+		o = s.newObject(false)
+		o.Mode = ModeRead
+	}
+
+	o.ID = filepath.Join(s.workDir, path)
+	o.Path = path
+	return o
+}
+func (s *Storage) createDir(ctx context.Context, path string) (o *Object, err error) {
+	rp := s.getAbsPath(path)
+	err = s.connection.ChangeDir(s.workDir)
```

Comment:
Is it necessary to change dir before `MakeDir`?

If it's required, we may need some lock for these operations, because users could call functions concurrently.